swim/memberlist.go

```diff
@@ -356,6 +386,11 @@ func (m *memberlist) Apply(change Change) {
 	member, ok := m.members.byAddress[change.Address]
 
 	if !ok {
+		// avoid indefinite tombstones by not creating new nodes
+		// directly in this state
```

Comment:
This code could be triggered in two cases.

1. When you join a cluster that has tombstones in them
2. When you receive a gossip for a tombstone after you have locally removed the member already.

If we would not have this here there is a chance for perpetual tombstones being gossiped around the system.